pkg/migrate/build-manifests.go

```diff
@@ -182,7 +184,12 @@ func (m *Manifest) FromHelmChart() error {
 			return errors.Newf("pod %s isn't scheduled to a node", podName)
 		}
 
-		nodeSpec := buildNodeSpecFromHelm(sts, pod.Spec.NodeName, input)
+		// Fetch the walFailoverSpec if applicable
+		walSpec := getWalFailoverSpec(ctx, m.clientset, sts, pod.Spec.NodeName, nodeIdx, input)
```

Comment:
We should add a field `walFailoverSpec` in the `parsedMigrationInput` struct. 
Rename the `func buildWalFailoverSpec` and keep the walFailoverSpec in the input.
This way we don't need to pass it separately in buildNodeSpecFromHelm.